azcli_conformity.sh

```diff
@@ -0,0 +1,60 @@
+# Define Variables for the Configurations
+export APP_SECRET=$(uuidgen)
+
+# Ask User Input for the App Registration Name
+echo " "
+echo "Which name do you want to give to your App Registration?"
+IFS=$'\n'
+read -r APP_NAME
+echo " "
+echo "Hello, Thank you for trying this script out.
+We're configuring your Azure Account to work with Cloud One Conformity, this might take several minutes depending of how many subscriptions you have."
+
+# Create App Registration
+az ad app create --display-name "$APP_NAME" --password $APP_SECRET --credential-description secret --required-resource-accesses @manifest.json > /dev/null
+
+# Export App Registration ID and Active Directory ID
+export AD_ID=`az account show --query tenantId --output tsv`
+export APPID=`az ad app list --display-name "$APP_NAME" | grep appId | cut -d ":" -f2 | grep -o '".*"' | sed 's/^"\(.*\)".*/\1/'`
+
+# Create a Service Principal for the App Registration
+az ad sp create --id $APPID > /dev/null
+
+# Grant Admin Consent to application
+az ad app permission admin-consent --id $APPID
```

Comment:
❓ When I run the script I get the following error.
```
Forbidden({"ClassName":"Microsoft.Portal.Framework.Exceptions.ClientException",
"Message":"Graph call failed with httpCode=Forbidden, errorCode=Authorization_RequestDenied, errorMessage=This operation can only be performed by an administrator. Sign out and sign in as an administrator or contact one of your organization's administrators., reason=Forbidden, correlationId = 1c3814c2-1d05-4315-9908-4c49811523f0, response = {\"odata.error\":{\"code\":\"Authorization_RequestDenied\",\"message\":{\"lang\":\"en\",\"value\":\"This operation can only be performed by an administrator. Sign out and sign in as an administrator or contact one of your organization's administrators.\"},\"requestId\":\"2b5dff55-205e-4ec6-8e28-3f9581e149f3\",\"date\":\"2020-09-28T09:40:34\"}}","Data":{},"HResult":-2146233088,"XMsServerRequestId":null,"Source":null,"HttpStatusCode":403,"ClientData":{"errorCode":"Authorization_RequestDenied","localizedErrorDetails":{"errorDetail":"This operation can only be performed by an administrator. Sign out and sign in as an administrator or contact one of your organization's administrators."},"operationResults":null,"timeStampUtc":"2020-09-28T09:40:34.4880801Z","clientRequestId":"1c3814c2-1d05-4315-9908-4c49811523f0","internalTransactionId":"1f82a7c1-7e05-4e59-a59a-582ba2717c6f","tenantId":"4e6682d3-a3e7-45be-b8a4-133ac2c6e747","userObjectId":"86fee386-adaf-4d97-a9f9-000ac0a14c8e","exceptionType":"AADGraphException"}})
```

I assumed that it was because I'm not an admin user however these are my permissions:

![image](https://user-images.githubusercontent.com/3625243/94513963-5c63e300-0263-11eb-91b3-179982b87e51.png)

Any ideas as to why this might be the case. Do I need higher permissions?

Reply:
From microsoft docs:
Granting tenant-wide admin consent requires you to sign in as Global Administrator, an Application Administrator, or a Cloud Application Administrator.

In your case, it appears you should have the required permissions already. One thing that may be potentially stopping is "Admin Consent Workflow (Preview)" if this has been set up then this may even restrict global admins from being able to register, if so we need to build a check for it also understand what impact that has on the overall onboarding. In terms of it being in the script, we are simply mimicking the steps documented as part of the manual setup which assumes they would have that access.

Reply:
Interesting. The plan is to try out the manual steps you have provided today or tomorrow so will be interesting to see if I can do it on the Portal.